Python/Product/Cookiecutter/CookiecutterPackage.cs

```diff
@@ -201,27 +208,18 @@ public sealed class CookiecutterPackage : Package, IOleCommandTarget {
             shell.ShowContextMenu(0, commandId.Guid, commandId.ID, pts, commandTarget);
         }
 
-        internal WindowPane ShowWindowPane(Type windowType, bool focus) {
-            var window = FindWindowPane(windowType, 0, true) as ToolWindowPane;
-            if (window != null) {
-                var frame = window.Frame as IVsWindowFrame;
-                if (frame != null) {
-                    ErrorHandler.ThrowOnFailure(frame.Show());
-                }
-
-                if (focus) {
-                    var content = window.Content as System.Windows.UIElement;
-                    if (content != null) {
-                        content.Focus();
-                    }
-                }
+        internal async Task<T> ShowWindowPaneAsync<T>(bool focus) where T : ToolWindowPane {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(DisposalToken);
```

Comment:
This will probably break mock tests (where we load packages), FYI. That's why we stick to using the UIThread class.

Reply:
Well, Cookiecutter tests pass 100%, but if we have mock tests that load packages, we may have problems with them anyway - `AsyncPackage` uses `JoinableTaskFactory` instance quite intensively

@huguesv , can you suggest a mock test that loads `PythonToolsPackage`?

Reply:
Not for cookiecutter, it doesn't use mock tests.

Reply:
Ok, what about other packages?

Reply:
I ran a random sample of PythonToolsMockTests (at least one for each test class) and didn't see PythonToolsPackage.Initialize being called.

Reply:
`PythonToolsMockTests` fail on my machine with MEF composition exception: https://github.com/Microsoft/PTVS/issues/3978, so I can't check it right now.

Reply:
It should compose fine - the test machines seem okay, so you probably need to fix it in part of the PR.

Reply:
I can repro it on a clean master after erasing `BuildOutput`. Probably need to update my VS to the recent build.